### PR TITLE
fix #8

### LINKED ISSUE #8: Patch files aren't up to date
Some findings : 

* Patching mariadb as we did for stable/mitaka don't work anymore
* Patching site.yml shouldn't be necessary anymore since it workarounds a limitation of ansible 1.9.x

Before diving into this we should think twice if the use the `node_custom_config` is relevant.

### LINKED ISSUE #36: Galera patch is outdated
I tested it with Kolla master and branch 3.0.1, each time mariadb fails to start.
So I assume `galera.cnf` needs an update.

### PR DESCRIPTION
**Pull request by msimonin**
closes #36

### CODE REVIEW

**Comment by msimonin** [comment]
To be tested.
3 patches need to be installed to deploy a standalone mariadb.

**Comment by msimonin** [comment]
Tested on vbox with : 
```
patches:
  - name: patch mariadb start
    src: mariadb_start.yml
    dst: kolla/ansible/roles/mariadb/tasks/start.yml
    enabled: "yes"
  - name: patch mariadb bootstrap
    src: mariadb_bootstrap.yml
    dst: kolla/ansible/roles/mariadb/tasks/bootstrap.yml
    enabled: "yes"
  - name: patch galera.cnf.j2
    src: galera.cnf.j2
    dst: kolla/ansible/roles/mariadb/templates/galera.cnf.j2
    enabled: "yes"
```

Hope this will helps

**Comment by msimonin** [comment]
Just added the doc on how to deploy from a dedicated node.

**Comment by msimonin** [comment]
rebased on master

File: enos/ansible/roles/patches/files/mariadb_start.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        ( groups['mariadb'] | length ) == 1 or
+        ( delegate_host == 'None' and mariadb_recover_inventory_name is not defined and inventory_hostname != groups['mariadb'][0] ) or
+        ( delegate_host == 'None' and mariadb_recover_inventory_name is defined and inventory_hostname != mariadb_recover_inventory_name )
+
+
+# TODO(jeffrey4l), remove the task check when the wair_for bug is fixed
+# https://github.com/ansible/ansible-modules-core/issues/2788
+- name: Waiting for MariaDB service to be ready
+  wait_for:
+    host: "{{ hostvars[inventory_hostname]['ansible_' + api_interface]['ipv4']['address'] }}"
```
Comment (rcherrueau) [inline enos/ansible/roles/patches/files/mariadb_start.yml]: should this be `{{ api_interface_address }}` (maybe kolla related, I don't know the original file)?

Comment (msimonin) [inline enos/ansible/roles/patches/files/mariadb_start.yml]: It's from the original file.

Comment (rcherrueau) [inline enos/ansible/roles/patches/files/mariadb_start.yml]: In this case, everything looks good to me

**Pull request merged**